mlflow/utils/autologging/__init__.py

```diff
@@ -422,18 +430,35 @@ def autolog(*args, **kwargs):
             config_to_store.update(kwargs)
             AUTOLOGGING_INTEGRATIONS[name] = config_to_store
 
-            try:
-                # Pass `autolog()` arguments to `log_autolog_called` in keyword format to enable
-                # event loggers to more easily identify important configuration parameters
-                # (e.g., `disable`) without examining positional arguments. Passing positional
-                # arguments to `log_autolog_called` is deprecated in MLflow > 1.13.1
-                AutologgingEventLogger.get_logger().log_autolog_called(name, (), config_to_store)
-            except Exception:
-                pass
+            is_silent_mode = get_autologging_config(name, "silent", False)
+            # Reroute non-MLflow warnings encountered during autologging enablement to an
+            # MLflow event logger, and enforce silent mode if applicable (i.e. if the corresponding
+            # autologging integration was called with `silent=True`)
+            with set_mlflow_events_and_warnings_behavior_globally(
+                # MLflow warnings emitted during autologging setup / enablement are likely
+                # actionable and relevant to the user, so they should be emitted as normal
+                # when `silent=False`
+                reroute_warnings=False,
+                disable_event_logs=is_silent_mode,
+                disable_warnings=is_silent_mode,
+            ), set_non_mlflow_warnings_behavior_for_current_thread(
+                reroute_warnings=True, disable_warnings=is_silent_mode,
+            ):
```

Comment:
For reference, https://docs.python.org/3/howto/logging.html#when-to-use-logging indicates that `warnings.warn()` should be used for avoidable issues that the client application / caller should address by modifying behaviors, while `logger.warning()` should be used to inform users of unexpected errors.

When enabling MLflow autologging, MLflow warnings may be emitted to inform callers of an issue. These are actionable & should be represented as warnings. non-MLflow warnings emitted during enablement are a byproduct of the MLflow autologging implementation and are therefore not actionable for the user, so we reroute them to `logger.warning()`.